stdlib/jdbc/src/main/java/org/ballerinax/jdbc/statement/UpdateStatement.java

```diff
@@ -70,26 +71,36 @@ public Object execute() {
         checkAndObserveSQLAction(strand, datasource, query);
         boolean isInTransaction = strand.isInTransaction();
         String errorMessagePrefix = "failed to execute update query: ";
+        MapValue<String, Object> generatedKeys = new MapValueImpl<>();
         try {
             ArrayValue generatedParams = constructParameters(parameters);
             conn = getDatabaseConnection(strand, client, datasource);
             String processedQuery = createProcessedQueryString(query, generatedParams);
-            stmt = conn.prepareStatement(processedQuery, Statement.RETURN_GENERATED_KEYS);
+
+            if (returnGeneratedKeys) {
+                stmt = conn.prepareStatement(processedQuery, PreparedStatement.RETURN_GENERATED_KEYS);
+            } else {
+                stmt = conn.prepareStatement(processedQuery);
+            }
+
             ProcessedStatement processedStatement = new ProcessedStatement(conn, stmt, generatedParams,
                     datasource.getDatabaseProductName());
             stmt = processedStatement.prepare();
             int count = stmt.executeUpdate();
-            MapValue<String, Object> generatedKeys;
-            if (!isDdlStatement()) {
-                rs = stmt.getGeneratedKeys();
-                //This result set contains the auto generated keys.
-                if (rs.next()) {
-                    generatedKeys = getGeneratedKeys(rs);
+            if (returnGeneratedKeys) {
+                if (isKeyRetrievalSupported()) {
```

Comment:
Why do we need this extra check?

Reply:
The returnGeneratedKeys only support INSERT, UPDATE, DELETE and MERGE operations So we need to check if the statement has one of these.

Reply:
Do we need specifically handle this? what if we not handled and user set return generated keys for other operation. do we get a meaningful error from the driver?

Reply:
We haven't got a meaningful error from the driver but got an empty ResultSet. So, I specifically handled this.

Reply:
@daneshk I too think we don't need to explicitly check this. Isn't it a driver or DB dependent thing whether throwing an exception or returning empty result set. Why we are limiting only to the `INSERT, DELETE, UPDATE, MERGE` since the developer now has the full control to retrieve the generated keys or not.

Reply:
Removed it.